sdk/resource/resource_test.go

```diff
@@ -183,7 +183,7 @@ func TestMerge(t *testing.T) {
 			name:  "Merge with different schemas",
 			a:     resource.NewWithAttributes("https://opentelemetry.io/schemas/1.4.0", kv41),
 			b:     resource.NewWithAttributes("https://opentelemetry.io/schemas/1.3.0", kv42),
-			want:  nil,
+			want:  []attribute.KeyValue{kv42},
```

Comment:
Can we add `wantErr error` field and set it here to `resource.ErrSchemaURLConflict`.

We can then change the assertion logic to e.g.

```go
switch {
case tt.wantErr != nil:
	require.ErrorIs(t, err, tt.wantErr)
case tt.isErr:
	require.Error(t, err)
default:
	require.NoError(t, err)
}
```

Reply:
updated.